packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2428,6 +2446,22 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     );
   }
 
+  /// Launch a web search on the current selection, as in the "Search Web" edit menu button on iOS.
+  /// Currently this is only implemented for iOS.
+  /// Throws an error if the selection is empty or collapsed.
+  Future<void> searchWebForSelection(SelectionChangedCause cause) async {
+    assert(!widget.obscureText);
+
+    final String text = textEditingValue.selection.textInside(textEditingValue.text);
+    if (widget.obscureText || text.isEmpty) {
+      return;
+    }
+    await SystemChannels.platform.invokeMethod(
```

Comment:
Nit: I think you could also remove the `async` from this method and instead return the result of this call instead?  Just calling out that as an option to consider, it's up to you if that's any better or not.

Reply:
That makes sense, but Im using async for both this function and the corresponding Look Up equivalent because Im following the pattern that `pasteText` uses  and I wanted to be consistent - is there a benefit to not using async in this case?

Reply:
No benefit that I'm aware of, I would keep it as-is in that case 👍